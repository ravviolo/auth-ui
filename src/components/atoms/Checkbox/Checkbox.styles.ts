import styled from 'styled-components';

import { colors } from 'theme';

export const ChechboxInput = styled.input`
  &[type='checkbox'] {
    accent-color: ${colors.primary};
  }
`;
