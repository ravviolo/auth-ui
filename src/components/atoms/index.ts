export { Button } from './Button/Button';
export { Checkbox } from './Checkbox/Checkbox';
export { Divider } from './Divider/Divider';
export { Heading } from './Heading/Heading';
export { IconButton } from './IconButton/IconButton';
export { InputField } from './InputField/InputField';
export { InputLabel } from './InputLabel/InputLabel';
export { TextButton } from './TextButton/TextButton';
