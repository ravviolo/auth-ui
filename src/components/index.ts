export * from './templates/LoginTemplate/LoginTemplate';
export * from './templates/SignUpTemplate/SignUpTemplate';
