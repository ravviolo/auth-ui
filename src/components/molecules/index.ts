export { AuthFooter } from './AuthFooter/AuthFooter';
export { AuthHeader } from './AuthHeader/AuthHeader';
export { AuthSocials } from './AuthSocials/AuthSocials';
export { EmailInput } from './EmailInput/EmailInput';
export { PasswordInput } from './PasswordInput/PasswordInput';
