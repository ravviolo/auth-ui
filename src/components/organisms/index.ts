export * from './LoginForm/LoginForm';
export * from './SignUpForm/SignUpForm';
