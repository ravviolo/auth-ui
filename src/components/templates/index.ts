export * from './LoginTemplate/LoginTemplate';
export * from './SignUpTemplate/SignUpTemplate';
