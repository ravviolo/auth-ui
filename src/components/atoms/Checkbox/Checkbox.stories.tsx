import { ComponentMeta, ComponentStory } from '@storybook/react';

import { Checkbox } from 'components/atoms';

export default {
  title: 'Atoms/Checkbox',
  component: Checkbox,
  args: {
    id: 'testCheckbox',
    name: 'testCheckbox',
  },
} as ComponentMeta<typeof Checkbox>;

export const Default: ComponentStory<typeof Checkbox> = (args) => (
  <Checkbox {...args} />
);
Default.storyName = 'Checkbox';
