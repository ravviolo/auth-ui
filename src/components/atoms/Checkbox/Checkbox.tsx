import React from 'react';

import * as S from './Checkbox.styles';

export interface Props {
  id: string;
  name: string;
  testId: string;
}

export const Checkbox = React.forwardRef<HTMLInputElement, Props>(
  ({ id, name, testId }, ref) => {
    return (
      <S.CheckboxInput
        ref={ref}
        data-testid={testId}
        id={id}
        name={name}
        type="checkbox"
      />
    );
  }
);

Checkbox.displayName = 'Checkbox';
