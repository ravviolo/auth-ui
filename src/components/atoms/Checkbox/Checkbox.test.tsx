import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { create } from 'react-test-renderer';

import { Checkbox } from 'components/atoms';

import { Props as CheckboxProps } from './Checkbox';

describe('Checkbox', () => {
  const props: CheckboxProps = {
    id: 'test-checkbox',
    name: 'test-checkbox',
    testId: 'checkbox-test-id',
  };

  it('should initially render unchecked', () => {
    render(<Checkbox {...props} />);

    const checkbox = screen.getByTestId('checkbox-test-id');

    expect(checkbox).not.toBeChecked();
  });

  it('should toggle checked state', async () => {
    render(<Checkbox {...props} />);

    const checkbox = screen.getByTestId('checkbox-test-id');

    await userEvent.click(checkbox);

    expect(checkbox).toBeChecked();

    await userEvent.click(checkbox);

    expect(checkbox).not.toBeChecked();
  });

  it('should match snapshot', () => {
    const tree = create(<Checkbox {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
