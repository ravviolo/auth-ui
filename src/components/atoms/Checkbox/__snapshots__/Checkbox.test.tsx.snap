// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should match snapshot 1`] = `
<input
  className="sc-gsnTZi kVuJZl"
  data-testid="test-checkbox"
  id="test-checkbox"
  name="test-checkbox"
  type="checkbox"
/>
`;
